lib/logstash/outputs/opensearch/http_client.rb

```diff
@@ -324,6 +324,12 @@ def build_adapter(options)
 
       adapter_options[:headers] = client_settings[:headers] if client_settings[:headers]
 
+      adapter_options[:auth_type] = options[:auth_type]
+
+      adapter_options[:port] = options[:port]
```

Comment:
Why is port and protocol being added as options?

Reply:
We are defining default values for port(443) and protocol("https") in api_configs.rb and passing it as options to manticore_adapter as it is being used to define the url if auth_type is "aws_iam". 

In AES, they are actually comparing the port number defined in the host(output configuration) with the default port passed in options and if there is a mismatch they are throwing an exception in http_client.rb before even doing a health check. But, in our case as we are still using localhost:9200, we are not doing this comparison. But, adding this in options will help us to add this  comparison if required in the future.

Reply:
i believe that user will be able to pass multiple url with and without port. if port number is specified in url then it will be used, else hostname:default port will be used. Shall we keep it this way? and, import only credentials/signing from amazon_es since validation of url should be outside scope of auth type.

Reply:
If the port number is not set in the host then it is considering 9200 as default port and the api call is failing for aws. Passing port number in options doesn't impact validation of master credentials and it is only used for AWS. 

I can even hard code them in manticore_adapter and remove it from options but the reason behind adding it is to compare mismatch if required, which I have explained in my above comment. If you think we may not do the comparison then we are good to remove it from options.

Reply:
If any config that will be used only by aws iam, it should be inside auth type. Though i am not convinced about hardcoding, if it is required for backward compatibility then i am fine with having DEFAULT_AWS_PORT as constant.

Reply:
You are right. But, I can't include it inside auth_type as we are also using auth_type for basic_auth and we don't use this port 443 for basic_auth. I will define it as a constant in manticore_adapter.